### DISCUSSION TITLE
An assembly specified in the application dependencies manifest (testhost.deps.json) was not found

### DISCUSSION THREAD

**Discussion by sandeeptalabathula** [danielpalme/ReportGenerator#710] [Q&A]
Following this integration guide - https://reportgenerator.io/getstarted

This command is continuously producing the below error. We don't 
PS C:\TestFolder> dotnet test --collect:"XPlat Code Coverage" "\users\YOUR_USERNAME\.nuget\packages\reportgenerator\5.4.1\tools\net8.0\ReportGenerator.exe" -reports:*\TestResults\*\coverage.cobertura.xml -targetdir:*\coveragereport

----------------------------
VSTest version 17.11.1 (x64)

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
Testhost process for source(s) '\users\YOUR_USERNAME\.nuget\packages\reportgenerator\5.4.1\tools\net8.0\ReportGenerator.exe' exited with error: Error:
  An assembly specified in the application dependencies manifest (testhost.deps.json) was not found:
    package: 'Newtonsoft.Json', version: '13.0.1'
    path: 'lib/netstandard2.0/Newtonsoft.Json.dll'
. Please check the diagnostic logs for more information.

Test Run Aborted.

I even tried to downgrade Newtonsoft.Json but that didn't help.

**Comment by danielpalme** [comment]
Which command does actually cause the error?

Is it the first line:
```
dotnet test --collect:"XPlat Code Coverage"
```

Or the second:
```
"\users\YOUR_USERNAME.nuget\packages\reportgenerator\5.4.1\tools\net8.0\ReportGenerator.exe" -reports:\TestResults*\coverage.cobertura.xml -targetdir:\coveragereport
```

If the first line? If yes, then I probably can't help. Maybe you have to update the `Microsoft.NET.Test.Sdk` dependency?
ReportGenerator does not have a dependency to `Newtonsoft.Json`.

**Reply by sandeeptalabathula** [reply]
Hey @danielpalme  Thanks for your prompt response. The first line works perfectly fine from Developer PowerShell. It gives output with all the passed tests along with` TestResults\<GUID>\coverage.cobertura.xml` files. 
The problem is when I run with the second line. I even tried to update `Microsoft.NET.Test.Sdk` to the current latest i.e.`17.11.1` but that didn't work. 

Here is the test project looks like if this helps...
```<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <IsPackable>false</IsPackable>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="coverlet.msbuild" Version="6.0.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="SomeInternalCustomLibrary" Version="1.0.23" />
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.4.1" />
    <PackageReference Include="Moq" Version="4.20.70" />
    <PackageReference Include="xunit" Version="2.9.0" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.8.2">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="coverlet.collector" Version="6.0.2">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
    <PackageReference Include="ReportGenerator" Version="5.4.1" />
  </ItemGroup>
</Project>```

**Comment by danielpalme** [comment]
I use the same commands in my projects and it works fine for me.

The error message seem strange to me:
```
An assembly specified in the application dependencies manifest (testhost.deps.json) was not found:
package: 'Newtonsoft.Json', version: '13.0.1'
path: 'lib/netstandard2.0/Newtonsoft.Json.dll'
```

ReportGenerator does not have a dependency to `Newtonsoft.Json`.

I have no idea what's wrong with your setup.
If you like, you can create a minimum sample, then I can have a closer look.
